cmd/serve/main.go

```diff
@@ -239,11 +223,377 @@ func run() (err error) {
 	return nil
 }
 
+func newDB(logger *slog.Logger) (*gorm.DB, error) {
+	host, err := requireEnv("DATABASE_HOST")
+	if err != nil {
+		return nil, err
+	}
+	port, err := requireEnvAsInt("DATABASE_PORT")
+	if err != nil {
+		return nil, err
+	}
+	username, err := requireEnv("DATABASE_USERNAME")
+	if err != nil {
+		return nil, err
+	}
+	password, err := requireEnv("DATABASE_PASSWORD")
+	if err != nil {
+		return nil, err
+	}
+	name, err := requireEnv("DATABASE_NAME")
+	if err != nil {
+		return nil, err
+	}
+
+	db, err := storage.NewDatabase(
+		logger,
+		storage.PostgresqlConfig{Host: host, Port: port, Username: username, Password: password, DatabaseName: name},
```

Comment:
The variables already have sane names so there's no reason to list the struct keys

```suggestion
		storage.PostgresqlConfig{host, port, username, password, name},
```